src/launcher/utils.cxx

```
@@ -0,0 +1,91 @@
+#include <windows.h>
+#include <WinInet.h>
+#include <iostream>
+#include <sstream>
+#include <time.h>
+
+#include "utils.hxx"
+
+const int RESPONSE_DATA_BUFFER_SIZE = 10000;
+
+//https://stackoverflow.com/a/49941336
+time_t ParseISO8601(const std::string& input)
+{
+    constexpr const size_t expectedLength = sizeof("1234-12-12T12:12:12Z") - 1;
+    static_assert(expectedLength == 20, "Unexpected ISO 8601 date/time length");
+
+    if (input.length() < expectedLength)
+    {
+        return 0;
+    }
+
+    std::tm time = { 0 };
+    time.tm_year = std::strtol(&input[0], nullptr, 10) - 1900;
+    time.tm_mon = std::strtol(&input[5], nullptr, 10) - 1;
+    time.tm_mday = std::strtol(&input[8], nullptr, 10);
+    time.tm_hour = std::strtol(&input[11], nullptr, 10);
+    time.tm_min = std::strtol(&input[14], nullptr, 10);
+    time.tm_sec = std::strtol(&input[17], nullptr, 10);
+    time.tm_isdst = 0;
+    return mktime(&time);
+}
+
+// GET HTTPS request. Not asynchronous!
+// Largely inspired by https://stackoverflow.com/q/18910463
+std::wstring GetHttpsRequest(const std::wstring& _server,
+    const std::wstring& _page,
+    const std::wstring& _useragent,
+    const std::wstring& _params)
+{
+    // fat buffer
+    char responseDataBuffer[RESPONSE_DATA_BUFFER_SIZE] = { 0 };
+
+    // initialize WinInet
+    HINTERNET hInternet = ::InternetOpen(_useragent.c_str(), INTERNET_OPEN_TYPE_PRECONFIG, NULL, NULL, 0);
+    if (hInternet != NULL)
+    {
+        // open HTTP session
+        HINTERNET hConnect = ::InternetConnect(hInternet, _server.c_str(), INTERNET_DEFAULT_HTTPS_PORT, NULL, NULL, INTERNET_SERVICE_HTTP, NULL, 1);
+        if (hConnect != NULL)
+        {
+            std::wstring request = _page +
+                (_params.empty() ? L"" : (L"?" + _params));

```

Nit: flipping these conditionals reduces our indentation. I find this easier to read, but whatever works for you:
```
HINTERNET hInternet = ::InternetOpen(...);
if (hInternet == NULL) {
    // MessageBox(...failure details here...)
    return null;
}

// open HTTP session
HINTERNET hConnect = ::InternetConnect(...);
if (hConnect == NULL){
    // MessageBox(...failure details here...)
    return null;
}

std::wstring request = _page + (...);
```

---

https://github.com/adanducci/GGPOPLUSR/pull/52/commits/1f7e53cef78f0d7f17e2d67f2e81a436b3066ca1 should resolve this comment.
I return empty strings instead of messagesboxes and nulls to make it fail silently and have Launcher.exe continue